src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -101,6 +101,15 @@ private static void WriteClientFields(CodeWriter writer, LowLevelClient client)
                 .WriteXmlDocumentationSummary($"The HTTP pipeline for sending and receiving REST requests and responses.")
                 .Line($"public virtual {typeof(HttpPipeline)} Pipeline => {client.Fields.PipelineField.Name};");
 
+            var uriField = client.Fields.GetFieldByParameter("endpoint", new CSharpType(typeof(Uri)));
```

Comment:
You are passing only `Uri` type, endpoint can be string type. It is working because we are not using the type in [this](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/LowLevel/Output/ClientFields.cs#L97) method to get the endpoint field. Can we call [this](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/LowLevel/Output/ClientFields.cs#L105) method instead?

Reply:
Shivangi and I discussed this offline - she is going to understand the reason why string is allowed for `endpoint` in some cases and whether that's the behavior we want.

Reply:
There is no real concept of Uri in Swagger, it's all string with the format metadata. [Here](https://github.com/Azure/autorest.csharp/tree/c43fd0f55fbb7cf0b871eeb7154f87ed8d4e6909/samples/Azure.Analytics.Purview.Account#model-endpoint-parameter-as-a-url-not-a-string) is one example of how we change from string to Uri type.

In terms of generator we don't want to change anything now. Maybe in future we will add analyzer for this and we could say something like: "If a client level path parameter is used in a parametrized host, then it's an endpoint and it must be format as url"